roles/slurm-management/tasks/main.yml

```diff
@@ -0,0 +1,227 @@
+    - restart_slurmctld
+  become: true
+
+- name: Install munge.key file.
+  copy:
+    src: "files/{{ slurm_cluster_name }}_munge.key"
+    owner: 'munge'
+    group: 'munge'
+    mode: '0600'
+    dest: '/etc/munge/munge.key'
+  notify:
+    - restart_munge
+    - restart_slurmdbd
+    - restart_slurmctld
+  become: true
+
+- name: Make sure the database root user is present and grant privileges.
+  mysql_user:
+    login_host: 127.0.0.1
+    login_user: 'root'
+    login_password: "{{ MYSQL_ROOT_PASSWORD }}"
+    name: "{{ slurm_storage_user }}"
+    password: "{{ slurm_storage_pass }}"
+    host: '%'
+    priv: '*.*:ALL'
+  no_log: True
+
+- name: Create a database for Slurm accounting.
+  mysql_db:
+    login_host: 127.0.0.1
+    login_user: 'root'
+    login_password: "{{ MYSQL_ROOT_PASSWORD }}"
+    name: 'slurm_acct_db'
+    state: 'present'
+  no_log: True
+
+- name: Create folders for Slurm.
+  file:
+    name: "{{ item.name }}"
+    owner: "{{ item.owner }}"
+    group: "{{ item.group }}"
+    mode: "{{ item.mode }}"
+    state: directory
+  with_items:
+    - name: '/etc/slurm'
+      owner: 'root'
+      group: 'root'
+      mode: '0755'
+    - name: '/var/log/slurm'
+      owner: 'slurm'
+      group: 'root'
+      mode: '0750'
+    - name: '/var/spool/slurm'
+      owner: 'slurm'
+      group: 'root'
+      mode: '0750'
+  become: true
+
+- name: Install Slurm config file.
+  template:
+    src: 'templates/slurm.conf'
+    dest: '/etc/slurm/slurm.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
+  notify: reload_slurmctld
+  become: true
+
+- name: Install Slurm DBD config file.
+  template:
+    src: 'templates/slurmdbd.conf'
+    dest: '/etc/slurm/slurmdbd.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
+  notify: reload_slurmdbd
+  become: true
+
+- name: Install Slurm scripts.
+  copy:
+    src: "files/{{ item }}"
+    dest: "/etc/slurm/{{ item }}"
+    owner: 'root'
+    group: 'root'
+    mode: '0755'
+  with_items:
+    - 'job_submit.lua'
+  become: true
+
+- name: Make sure slurmdbd and munge services are enabled and started.
+  systemd:
+    name: "{{item}}"
+    state: 'started'
+    enabled: 'yes'
+  with_items:
+    - 'munge.service'
+    - 'slurmdbd.service'
+  become: true
+
+#
+# ToDo: makebash script idempotent.
+#
+- name: Copy Slurm DB initialization script to SAI.
+  template:
+    src: 'templates/configure_slurm_accounting_db.bash'
+    dest: '/etc/slurm/configure_slurm_accounting_db.bash'
+    owner: 'root'
+    group: 'root'
+    mode: '0700'
+  become: true
+
+#
+# Flush handlers to make sure Slurm DB is present and connection settings have been updated.
+#
+- meta: flush_handlers
+
+- name: Execute Slurm DB initialization script on SAI.
+  shell: |
+         /etc/slurm/configure_slurm_accounting_db.bash
+  tags:
+    - create_database
+  register: command_result
+#  failed_when: >
+#    command_result.rc != 0
+#    and "already exists" not in command_result.stdout
```

Comment:
please remove

Reply:
Not removed, but improved: failure to configure the Slurm accounting DB is now properly detected using ```failed_when```.